appservice/src/intTest/groovy/nu/yona/server/UserPhotoTest.groovy

```diff
@@ -0,0 +1,290 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Stichting Yona Foundation
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v.2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server
+
+import org.apache.http.HttpEntity
+import org.apache.http.entity.mime.MultipartEntityBuilder
+import org.apache.http.entity.mime.content.InputStreamBody
+import org.codehaus.groovy.runtime.MethodClosure
+
+import groovy.json.*
+import nu.yona.server.test.User
+
+class UserPhotoTest extends AbstractAppServiceIntegrationTest
+{
+	def 'User uploads user photo'()
+	{
+		given:
+		def richard = addRichard()
+		appService.yonaServer.restClient.encoder.putAt("multipart/form-data", new MethodClosure(this, 'encodeMultipartEntity'))
+		def multipartEntity = MultipartEntityBuilder.create()
+				.addPart("file", new InputStreamBody(new ByteArrayInputStream(Base64.getDecoder().decode("R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7")), "image/gif", "MyPhoto.png"))
```

Comment:
Not sure what the bytes represent, but I suggest keeping the mime type equal to the file extension, so both gif or both png, preferably in line with the bytes.

Reply:
Good point, will add a PNG example